mailpoet/assets/js/src/segments/dynamic/form.tsx

```diff
@@ -82,130 +81,136 @@ export function Form({ segmentId }: Props): JSX.Element {
   );
 
   return (
-    <form className="mailpoet_form">
-      {errors.length > 0 && (
-        <APIErrorsNotice errors={errors.map((error) => ({ message: error }))} />
-      )}
-      <div className="mailpoet-form-grid">
-        <div className="mailpoet-form-field-name form-field-row-name mailpoet-segments-name-section">
-          <Heading level={4}>
-            <label htmlFor="field_name">{__('Name', 'mailpoet')}</label>
-          </Heading>
-          <div className="mailpoet-form-field">
+    <div className="mailpoet-form-container">
+      <form>
+        {errors.length > 0 && (
+          <APIErrorsNotice
+            errors={errors.map((error) => ({ message: error }))}
+          />
+        )}
+        <FieldsSection
+          title={__('Details', 'mailpoet')}
+          description={__('What do you want to call this segment?', 'mailpoet')}
+        >
+          <FieldWrapper slug="name" title={__('Name', 'mailpoet')}>
```

Comment:
I agree that this solution is not great precisely because of the reason that you pointed out. That being said, if I understand your suggestion, I'm not sure if it would be much better because then we would need to create one component for each of the calls to `<FieldWrapper>`, no? Or at least, for each of the calls that have different components as children. This would remove the flexibility and reusability of `<FieldWrapper>`. Maybe we can ask other developers if they have another suggestion? I can't think of anything with my limited React knowledge.

Reply:
Yes, my idea would require creating a new component for each configuration of children, which I agree wouldn't be much better.

I may not be fully understanding the idea behind `FieldWrapper`. Right now it's being used for the name, description, and conditions on the segment editor page. In the case of the [conditions](https://github.com/mailpoet/mailpoet/blob/3e7399c440f6bd5c6d84b558f1954bc364e7bb2e/mailpoet/assets/js/src/segments/dynamic/form.tsx#L141-L141), it doesn't seem like it's doing much. Removing it doesn't have a visual effect that I can see, and although you're passing it a `slug` I don't think it gets used because it's not receiving a `title`. 

Some combinations also might not make sense. For example, `FieldWrapper` will render multiple children, but it would never make sense to give it multiple `Input`s because it only ever renders a single label/description.

Right now it seems like the wrapper is mostly useful as a wrapper around text inputs, which is why I think it feels simpler to create something more specific like a `TextInputField` that would include the wrapper and the input, but again I think I don't understand the idea for how it should be used more generally.

This could definitely be a good thing to bring up with the group, especially if this is a component we'll be using a lot in newly designed pages.

Reply:
> I may not be fully understanding the idea behind FieldWrapper.

My goal with creating <FieldWrapper> was to create a component that, together with <FieldsSection>, would make it easier to recreate the same design for similar pages.

> In the case of the conditions, it doesn't seem like it's doing much.

I agree, and this is something that I hadn't considered.

> Removing it doesn't have a visual effect that I can see, and although you're passing it a slug I don't think it gets used because it's not receiving a title.

Good point, I failed to notice this when I made title optional. I updated the component to make slug optional as well.

> For example, FieldWrapper will render multiple children, but it would never make sense to give it multiple Inputs because it only ever renders a single label/description.

I'm not sure if there is a way to limit the number of children that can be passed to this component. In the future, I think we can improve it to accept a type of input field and then the component itself will create the field.

> Right now it seems like the wrapper is mostly useful as a wrapper around text inputs, which is why I think it feels simpler to create something more specific like a TextInputField that would include the wrapper and the input, but again I think I don't understand the idea for how it should be used more generally.

I assumed it could be used as well for other types of inputs, like a single select box. But I'm not sure as we currently don't have such a field.

I agree that the `<FieldWrapper>` abstraction needs to be improved and maybe even completely changed. My suggestion is that we leave it as is for now and reconsider it when it is time to implement another page with the same design. I feel like then we will have more data to judge what is the best path for this component.

Reply:
Thanks for your thoughtful reply! I agree with leaving it as is for now and modifying it as needed as we need to implement different designs.